modules/components/pages/inputs/pg_stream.adoc

```diff
@@ -0,0 +1,467 @@
+
+. Restart the PostgreSQL server.
+--
+=====
+
+== Choose a replication mode
+
+When you run a pipeline that uses the `pg_stream` input, Redpanda Connect connects to your PostgreSQL database and creates a replication slot. The replication slot uses a copy of the Write-Ahead Log (WAL) file to subscribe to changes in your database records as they are applied to the database. There are two replication modes you can choose from: snapshot mode and streaming mode.
+
+In snapshot mode, Redpanda Connect first takes a snapshot of the database and streams the contents before processing changes from the WAL. In streaming mode, Redpanda Connect directly processes changes from the WAL starting from the most recent changes without taking a snapshot first.
+
+For local testing, you can use the <<example-pipeline,example pipeline on this page>>, which runs in snapshot mode.
+
+=== Snapshot mode
+
+If you set the <<stream_snapshot,`stream_snapshot` field>> to `true`, Redpanda Connect:
+
+. Creates a snapshot of your database.
+. Streams the contents of the tables specified in the `pg_stream` input.
+. Starts processing changes in the WAL that occurred since the snapshot was taken, and streams them to Redpanda Connect. 
+
+Once the initial replication process is complete, the snapshot is removed and the input keeps a connection open to the database so that it can receive data updates.
+
+If the pipeline restarts during the replication process, Redpanda Connect resumes processing data changes from where it left off. If there are other interruptions while the snapshot is taken, you may need to restart the snapshot process. For more information, see <<troubleshoot_replication_failures,Troubleshoot replication failures>>.
+
+=== Streaming mode
+
+If you set the <<stream_snapshot,`stream_snapshot` field>> to `false`, Redpanda Connect starts processing data changes from the end of the WAL. If the pipeline restarts, Redpanda Connect resumes processing data changes from the last acknowledged position in the WAL.
+
+== Monitor the replication process
+
+You can monitor the initial replication of data using the following metrics:
+
+|===
+| Metric name | Description
+
+| `replication_lag_bytes`
+| Indicates how far the connector is lagging behind the source database when processing the transaction log.
+
+| `postgres_snapshot_progress`
+| Shows the progress of snapshot processing for each table.
+
+|===
+
+
+== Troubleshoot replication failures
+
+If the database snapshot fails, the replication slot has only an incomplete record of the existing data in your database. To maintain data integrity, you must drop the replication slot manually in your source database and run the Redpanda Connect pipeline again.
+
+```SQL
+SELECT pg_drop_replication_slot(SLOT_NAME);
+```
+
+== Receive TOAST and deleted values
+
+For full visibility of all data updates, you can also choose to stream TOAST and deleted values. To enable this option, run the following query on your source database:
```

Comment:
Lets add a link to define TOAST: https://www.postgresql.org/docs/current/storage-toast.html

Reply:
Hi @rockwotj. Added the link. Also, added your comment about networking from the related Cloud PR. You can see the preview of the page here: [https://deploy-preview-140--rp-cloud.netlify.app/redpanda-cloud/develop/connect/components/inputs/pg_stream/](https://deploy-preview-140--rp-cloud.netlify.app/redpanda-cloud/develop/connect/components/inputs/pg_stream/ )I've added a new prerequisite.